File: algorithms/1550.three-consecutive-odds/solution.rb
# @param {Integer[]} arr
# @return {Boolean}
def three_consecutive_odds(arr)
  arr.each_cons(3) do |a, b, c|
    return true if [a, b, c].count { |i| i % 2 == 0 } == 0
  end
  false
end

File: algorithms/1550.three-consecutive-odds/solution.rs
impl Solution {
    pub fn three_consecutive_odds(arr: Vec<i32>) -> bool {
        for w in arr.windows(3) {
            if [w[0], w[1], w[2]].iter().filter(|&i| i % 2 == 0).count() == 0 {
                return true;
            }
        }
        false
    }
}
